app/src/main/java/org/mozilla/focus/web/GeckoWebViewProvider.kt

```diff
@@ -597,7 +605,7 @@ class GeckoWebViewProvider : IWebViewProvider {
 
         override fun restoreWebViewState(session: Session) {
             val stateData = session.savedWebViewState!!
-            val savedSession = stateData.getParcelable<GeckoSession>(GECKO_SESSION)!!
+            val savedSession = session.savedGeckoSession!!
```

Comment:
Does this also work with "don't keep activities"? 🤔

Reply:
Hmm.. in a Custom Tab this fails gracefully by showing the Focus home screen, but on the main browser it doesn't. Looking into this..

Reply:
I tested this latest change over the week~end~, and with "don't keep activities" there is something wacky that happens when restoring the session where the session is correctly restored first, but our `NestedGeckoView` is created after and it holds a default state which then overwrites what the UI sees.

I'd say this is more reason to hop onto the AC Engine now instead of trying to fix this.